src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.MemOps.cs

```diff
@@ -0,0 +1,537 @@
+            Unsafe.WriteUnaligned<long>(ref Unsafe.Add(ref b, 16), 0);
+            Unsafe.WriteUnaligned<long>(ref Unsafe.Add(ref b, 24), 0);
+            Unsafe.WriteUnaligned<long>(ref Unsafe.Add(ref b, 32), 0);
+            Unsafe.WriteUnaligned<long>(ref Unsafe.Add(ref b, 40), 0);
+            Unsafe.WriteUnaligned<long>(ref Unsafe.Add(ref b, 48), 0);
+            Unsafe.WriteUnaligned<long>(ref Unsafe.Add(ref b, 56), 0);
+#else
+            Unsafe.WriteUnaligned<int>(ref b, 0);
+            Unsafe.WriteUnaligned<int>(ref Unsafe.Add(ref b, 4), 0);
+            Unsafe.WriteUnaligned<int>(ref Unsafe.Add(ref b, 8), 0);
+            Unsafe.WriteUnaligned<int>(ref Unsafe.Add(ref b, 12), 0);
+            Unsafe.WriteUnaligned<int>(ref Unsafe.Add(ref b, 16), 0);
+            Unsafe.WriteUnaligned<int>(ref Unsafe.Add(ref b, 20), 0);
+            Unsafe.WriteUnaligned<int>(ref Unsafe.Add(ref b, 24), 0);
+            Unsafe.WriteUnaligned<int>(ref Unsafe.Add(ref b, 28), 0);
+            Unsafe.WriteUnaligned<int>(ref Unsafe.Add(ref b, 32), 0);
+            Unsafe.WriteUnaligned<int>(ref Unsafe.Add(ref b, 36), 0);
+            Unsafe.WriteUnaligned<int>(ref Unsafe.Add(ref b, 40), 0);
+            Unsafe.WriteUnaligned<int>(ref Unsafe.Add(ref b, 44), 0);
+            Unsafe.WriteUnaligned<int>(ref Unsafe.Add(ref b, 48), 0);
+            Unsafe.WriteUnaligned<int>(ref Unsafe.Add(ref b, 52), 0);
+            Unsafe.WriteUnaligned<int>(ref Unsafe.Add(ref b, 56), 0);
+            Unsafe.WriteUnaligned<int>(ref Unsafe.Add(ref b, 60), 0);
+#endif
+            b = ref Unsafe.Add(ref b, 64);
+            n--;
+            if (n != 0)
+                goto MZER06;
+
+            byteLength %= 64;
+            if (byteLength > 16)
+                goto MZER00;
+#if HAS_CUSTOM_BLOCKS
+            Unsafe.WriteUnaligned<Buffer.Block16>(ref Unsafe.Add(ref bEnd, -16), default);
+#elif TARGET_64BIT
+            Unsafe.WriteUnaligned<long>(ref Unsafe.Add(ref bEnd, -16), 0);
+            Unsafe.WriteUnaligned<long>(ref Unsafe.Add(ref bEnd, -8), 0);
+#else
+            Unsafe.WriteUnaligned<int>(ref Unsafe.Add(ref bEnd, -16), 0);
+            Unsafe.WriteUnaligned<int>(ref Unsafe.Add(ref bEnd, -12), 0);
+            Unsafe.WriteUnaligned<int>(ref Unsafe.Add(ref bEnd, -8), 0);
+            Unsafe.WriteUnaligned<int>(ref Unsafe.Add(ref bEnd, -4), 0);
+#endif
+            return;
+
+        PInvoke:
+            Buffer._ZeroMemory(ref b, byteLength);
+        }
+
+#if NATIVEAOT
+        [System.Runtime.RuntimeExport("RhRuntimeHelpers_MemCopy")]
+#endif
+        private static unsafe void MemCopy(ref byte dest, ref byte src, nuint len)
+        {
+            if (len == 0)
+            {
+                return;
+            }
+
+            // Implicit nullchecks
+            _ = Unsafe.ReadUnaligned<byte>(ref dest);
+            _ = Unsafe.ReadUnaligned<byte>(ref src);
+
+            // P/Invoke into the native version when the buffers are overlapping.
```

Comment:
If this is only meant to be used by cpblk IL instruction, this check is unnecessary. Overlapping buffers are UB for cpblk IL instruction.

Reply:
Also, I do not think that the perf of cpblk IL instruction with unknown size or large sze is particularly important. I think it is fine to forward to the main Memmove instead of manually inlining it here.

Reply:
If you think it is worth manually inlining it, it is ok with me - but at least keep the two copies next to each other.